spec/spec_helper_acceptance.rb

```diff
@@ -16,40 +16,33 @@
 install_module_on(hosts)
 install_module_dependencies_on(hosts)
 
+def idempotent_apply(hosts, manifest, opts = {}, &block)
+  block_on hosts, opts do |host|
+    file_path = host.tmpfile('apply_manifest.pp')
+    create_remote_file(host, file_path, manifest + "\n")
+
+    puppet_apply_opts = { :verbose => nil, 'detailed-exitcodes' => nil }
+    on_options = { acceptable_exit_codes: [0, 2] }
+    on host, puppet('apply', file_path, puppet_apply_opts), on_options, &block
+    puppet_apply_opts2 = { :verbose => nil, 'detailed-exitcodes' => nil }
+    on_options2 = { acceptable_exit_codes: [0] }
+    on host, puppet('apply', file_path, puppet_apply_opts2), on_options2, &block
+  end
+end
+
 RSpec.configure do |c|
   # Readable test descriptions
   c.formatter = :documentation
 
   # detect the situation where PUP-5016 is triggered and skip the idempotency tests in that case
   # also note how fact('puppetversion') is not available because of PUP-4359
-  if fact('osfamily') == 'Debian' && fact('operatingsystemmajrelease') == '8' && shell('puppet --version').stdout =~ %r{^4\.2}
+  if os[:family] == 'Debian' && os[:release] == '8' && shell('puppet --version').stdout =~ %r{^4\.2}
     c.filter_run_excluding skip_pup_5016: true
   end
 
   # Configure all nodes in nodeset
   c.before :suite do
     run_puppet_access_login(user: 'admin') if pe_install? && (Gem::Version.new(puppet_version) >= Gem::Version.new('5.0.0'))
-    hosts.each do |host|
```

Comment:
Should this section be removed? 
This is necessary for i18n testing.

Reply:
re-added that section